File: app/lib/features/auth/data/models/auth_session_model.dart
import '../../domain/entities/auth_session.dart';
import 'auth_user_model.dart';

class AuthSessionModel extends AuthSession {
  const AuthSessionModel({
    required super.accessToken,
    required super.user,
  });

  factory AuthSessionModel.fromJson(Map<String, dynamic> json) {
    return AuthSessionModel(
      accessToken: json['accessToken'] as String,
      user: AuthUserModel.fromJson(json['user'] as Map<String, dynamic>),
    );
  }
}

File: panel/src/features/auth/data/models/AuthSessionModel.ts
import type { AuthSession } from '../../domain/entities/AuthSession';
import { AuthUserModel } from './AuthUserModel';

export class AuthSessionModel {
  static fromJson(json: Record<string, unknown>): AuthSession {
    return {
      accessToken: String(json.accessToken),
      user: AuthUserModel.fromJson(json.user as Record<string, unknown>),
    };
  }
}